src/main/java/org/dormitory/autobotsoccub/command/querycommand/RegisterCommand.java

```diff
@@ -5,26 +5,45 @@
 import org.dormitory.autobotsoccub.command.keyboard.KeyboardFactory;
 import org.dormitory.autobotsoccub.command.keyboard.Button;
 import org.dormitory.autobotsoccub.command.result.CommandResult;
+import org.dormitory.autobotsoccub.user.UserPool;
 import org.telegram.telegrambots.api.objects.Update;
+import org.telegram.telegrambots.api.objects.User;
 
-import static org.dormitory.autobotsoccub.command.keyboard.Button.UNREGISTER;
-
+import static org.dormitory.autobotsoccub.command.keyboard.Button.*;
 
 @AllArgsConstructor
 public class RegisterCommand implements InlineQueryCommand {
 
     private KeyboardFactory keyboardFactory;
 
+    private UserPool userPool;
+
     @Override
     public Button getCommandButton() {
         return Button.REGISTER;
     }
 
     @Override
     public CommandResult execute(Update update) {
-        return CommandResult.builder()
-                .replyMessage("You are registered for upcoming game!")
-                .keyboardMarkup(keyboardFactory.keyboardOf(UNREGISTER))
-                .build();
+        synchronized (userPool) {
```

Comment:
I think we can achieve more fine grained sync. Will discuss this

Reply:
Okay) Removed for a while